### PR TITLE
ROX-20125: new base snapshot

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

Update the database snapshot used for most tests to be based off 4.1.  4.1 was chosen because the upgrade test goes through a backward compatibility test and 4.1 is where we introduced that requirement for the data.  

There remains a specific test test `legacy_to_postgres` that still uses the 3.56 snapshot.  This is because the purpose of that test is solely to test going from RocksDB to Postgres.  That test will be removed with ROX-22872 as will the rest of the references to the 3.56 snapshot.  Once that is complete we should be able to begin the rest of the RocksDB removal beginning with the migrations. 

Note:  I could have probably made some of these changes more elegant but it didn't seem worth it since some of these things are going to be removed entirely anyway.

## Checklist
- [ ] Investigated and inspected CI test results
~- [ ] Unit test and regression tests added~
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

The changes are purely to tests.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: tests/upgrade/lib.sh [hunk 1]
```diff
@@ -115,12 +116,51 @@ deploy_earlier_central() {
...
+         --set central.exposure.loadBalancer.enabled=true \
+         --set system.enablePodSecurityPolicies=false \
+         --set central.image.tag="${EARLIER_TAG}" \
+         --set central.db.image.tag="${EARLIER_TAG}" \
+         --set scanner.image.tag="$(cat SCANNER_VERSION)" \
+         --set scanner.dbImage.tag="$(cat SCANNER_VERSION)"
+
+    # Installing this way returns faster than the scripts but everything isn't running when it finishes like with
+    # the scripts.  So we will give it a minute for things to get started before we proceed
+    sleep 60
```
Comment (SimonBaeumer) [inline tests/upgrade/lib.sh:144]: If waiting for Central is sufficient this should do the trick:

```suggestion
    kubectl rollout status -n stackrox deployment/central
```

**Review by SimonBaeumer — APPROVED** [review]
Nice! :ship:

**Review by md2119 — APPROVED** [review]
This looks good. I would recommend dropping the important bits for upgrade test logs and results somewhere. Once we merge this PR, we are essentially changing the upgrade path. Therefore, saving the evidence of the behavior in the confluence would be great for posterity because the CI logs are available only for a limited time.

**Pull request merged**